frontend/src/views/supp-allowances/SupplementaryFormView.vue

```diff
@@ -37,10 +37,38 @@ import IndigenousProgrammingAllowance from '@/components/supp-allowances/Indigen
 import SupportNeedsProgrammingAllowance from '@/components/supp-allowances/SupportNeedsProgrammingAllowance.vue'
 import TransportationAllowance from '@/components/supp-allowances/TransportationAllowance.vue'
 import { isEmpty } from 'lodash'
+import { useApplicationsStore } from '@/stores/applications'
+import { mapActions } from 'pinia'
+import ApplicationService from '@/services/applicationService'
+import alertMixin from '@/mixins/alertMixin'
+
+function findAndUpdateModel(suppApplications, modelToUpdate) {
+  const found = suppApplications.find((application) => application.supplementaryType === modelToUpdate.supplementaryType)
+  if (!found) {
+    //no existing application, a blank model is required
+    return modelToUpdate
+  } else {
+    //use the application from dynamics
+    return found
+  }
+}
+
+//should add more in depth checks as model for transport will have string/number
+function isModelEmpty(model) {
+  let modelData = { ...model }
+
+  delete modelData.supplementaryApplicationId
+  delete modelData.supplementaryType
+
+  return Object.values(modelData).some((value) => {
```

Comment:
yes - agreed it needs to be more flexible for other sections. 
The reason I chose .some over .every is because I know for each application, there is only one array - and I want the obj to be marked "empty" if all checkboxes are empty, but there is text in the 'other' textbox. 
This would delete the application even if the user wrote some text, then unselected the 'other' checkbox 

for the sake of getting this bit to QA I will leave it for now but will update it when I start on the next section

Reply:
If you've already analyzed the structure of the other application types then it's probably good to go. We will just test as we go.